### PR TITLE
fix(civisibility): fix cases where the git pack-objects fails due to a cross-device error

### PR DESCRIPTION
**Pull request by tonyredondo**
### What does this PR do?

Git can throw a cross device error if the temporal folder is in a different drive than the .git folder (eg. symbolic link); to handle this edge case, we first try with a temp folder and if we fail then we try in the working directory folder.

### Motivation

Improve the resilience in some edge cases

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] New code is free of linting errors. You can check this by running `./scripts/lint.sh` locally.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/civisibility/utils/git.go [hunk 1]
```diff
@@ -656,18 +656,38 @@ func CreatePackFiles(commitsToInclude []string, commitsToExclude []string) []str
...
+		}
 	}
 
-	// git pack-objects --compression=9 --max-pack-size={MaxPackFileSizeInMb}m "{temporaryPath}"
-	out, err := execGitStringWithInput(telemetry.PackObjectsCommandsType, objectsShasString,
-		"pack-objects", "--compression=9", "--max-pack-size="+strconv.Itoa(MaxPackFileSizeInMb)+"m", temporaryPath+"/")
 	if err != nil {
-		log.Warn("civisibility: error creating pack files: %s", err.Error())
+		log.Warn("civisibility: error creating pack files in %s: %s", temporaryPath, err.Error())
 		return nil
```
Comment (anmarchenko) [inline internal/civisibility/utils/git.go:691]: should we also log `out` here?

Comment (tonyredondo) [reply internal/civisibility/utils/git.go:691]: not required `execGitStringWithInput` will log the output in debug mode so we don't pollute the stdout

**Review by anmarchenko — APPROVED** [review]
